### ISSUE TITLE
Querying Hive partitioning parquets is slow

### ISSUE THREAD

**Issue by xqe2011** [paradedb/pg_analytics#173]
### What happens?

Recently, we tried this extensions instead of using a standalone duckdb instance. When we run a simple `SELECT` query on parquet files, it's 2-20 times slower than DuckDB.

Profiling method
`SELECT duckdb_execute($$SET enable_profiling='query_tree'$$);` and watch logs.

### To Reproduce

Query one field : `SELECT name FROM public.table1 where code1 = 3261 and code2 = '001' and code3 = '5204' and code4 = '1'`
`code1` and `code2` are partition fields.

DuckDB runs on cli 0.0190s
DuckDB runs in this extension 0.0291s
Total time of using this extension 0.513s
Query multi fields: `SELECT name, level, detxlen, detylen, downid  FROM public.table1 where code1 = 3261 and code2 = '001' and code3 = '5204' and code4 = '1'`
`code1` and `code2` are partition fields.

DuckDB runs on cli 0.0413s
DuckDB runs in this extension 0.037s
Total time of using this extension  0.552s

### OS:

Ubuntu Server 22.04.3

### ParadeDB Version:

paradedb/paradedb:16-v0.11.1

### Are you using ParadeDB Docker, Helm, or the extension(s) standalone?

ParadeDB Helm Chart

### Full Name:

Liu Qijie

### Affiliation:

Dongguan University of Technology

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include the code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configurations (e.g., CPU architecture, PostgreSQL version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by philippemnoel** [comment]
Thanks for opening! Would love your help with debugging this, or anyone else if willing to assist here :)

**Comment by kysshsy** [comment]
It seems the most of time is spent on duckdb-rs. Need to dive into duckdb-rs.
![partition](https://github.com/user-attachments/assets/639468bc-073b-40c4-ac61-fad261c731aa)

**Comment by kysshsy** [comment]
@xqe2011 Hi, could you set the parameters and then test it again? Currently, I am not using the ParadeDB Helm Chart; I am directly using pg_analytics. Thank you!

**Comment by philippemnoel** [comment]
Thank you for investigating this @kysshsy. @xqe2011 if you are using the Helm chart, please be sure to allocate enough resources.

@kysshsy Do you think we should close this issue if there isn't anything on our end?

**Comment by kysshsy** [comment]
> Thank you for investigating this [@kysshsy](https://github.com/kysshsy). [@xqe2011](https://github.com/xqe2011) if you are using the Helm chart, please be sure to allocate enough resources.
> 
> [@kysshsy](https://github.com/kysshsy) Do you think we should close this issue if there isn't anything on our end?

Yeah, I think so. If the user provides more information indicating that the issue is not with DuckDB, we can reopen the issue.

**Comment by philippemnoel** [comment]
Thank you for investigating <3
